packages/react-reconciler/src/ReactFiberPendingWork.js

```diff
@@ -0,0 +1,243 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {FiberRoot} from './ReactFiberRoot';
+import type {ExpirationTime} from './ReactFiberExpirationTime';
+
+import {NoWork} from './ReactFiberExpirationTime';
+
+// Because we don't have a global queue of updates, we use this module to keep
+// track of the pending levels of work that have yet to be flushed. You can
+// think of a PendingWork object as representing a batch of work that will
+// all flush at the same time. The actual updates are spread throughout the
+// update queues of all the fibers in the tree, but those updates have
+// priorities that correspond to a PendingWork batch.
+
+export type PendingWork = {
+  // We use `expirationTime` to represent both a priority and a timeout. There's
+  // no inherent reason why they need to be the same, and we may split them
+  // in the future.
+  startTime: ExpirationTime,
+  expirationTime: ExpirationTime,
+  isSuspended: boolean,
+  shouldTryResuming: boolean,
+  isRenderPhaseWork: boolean,
+  next: PendingWork | null,
+};
+
+function insertPendingWorkAtPosition(root, work, insertAfter, insertBefore) {
+  work.next = insertBefore;
+  if (insertAfter === null) {
+    root.firstPendingWork = work;
+  } else {
+    insertAfter.next = work;
+  }
+}
+
+export function addPendingWork(
```

Comment:
The terms "add" and "work" is an implementation detail. `schedule` or `inform` might be better verbs. "Priority" is better than work.

We're not scheduling any "Work" which in other places refers to some computation.